pxtlib/browserutils.ts

```diff
@@ -143,6 +143,10 @@ namespace pxt.BrowserUtils {
         return isPxtElectron() || isIpcRenderer();
     }
 
+    export function isArcade() {
```

Comment:
typically we've avoided target specific checks in pxt core (e.g. there's one in `field_soundEffect.ts`, but with a fixme associated with it -- if possible, basing off the fact that there are hardware variants and one hasn't been selected yet, or adding a separate 'prompt for hardware selection' pxtarget flag worst case may be better.

Reply:
So you're saying that I don't need the extra arcade check? The "no hardware variant chosen" check should be sufficient enough?

Reply:
ideally, if we cloned arcade repo & just changed id, behavior would be the same. Looking at it that seems safe with `!pxt.getActiveHwVariant`, though maybe need a check on `pxt.appTarget.variants` being defined, so that it doesn't occur if there *aren't* variants, if the current check doesn't account for that.

Reply:
Ah, that's a good point, I should check for variants before checking for the active variant. That said, I don't know if just checking for a variants object is going to be enough to differentiate in this scenario. I don't think we'd want to pop open hardware dialog in microbit, but microbit also has a variants object for dal vs codal.

Reply:
I'd rather not create another flag, but I think I found a flag that would actually work well in this situation. Microbit has a different connect device experience because we have the `downloadDialogTheme` defined there. If we have a download theme defined, then we wouldn't want to pop up the hardware choosing option, we would always want to show the defined experience. So, in addition to checking that variants exist, I will also check that a download theme doesn't exist before showing the hardware picker modal.

Reply:
that sounds like would make sense yeah; this `isArcade` check is just a target flag that doesn't live in pxtarget, so any way you can differentiate is probably reasonable enough for this use case / can be adjusted in future if need be.